// ion-rating-stars/esm2020/lib/ion-rating-stars.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { IonicModule } from '@ionic/angular';
import { IonRatingStarsComponent } from './ion-rating-stars.component';
import * as i0 from "@angular/core";
export class IonRatingStarsModule {
}
IonRatingStarsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.7", ngImport: i0, type: IonRatingStarsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
IonRatingStarsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.7", ngImport: i0, type: IonRatingStarsModule, declarations: [IonRatingStarsComponent], imports: [CommonModule,
        IonicModule], exports: [IonRatingStarsComponent] });
IonRatingStarsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.7", ngImport: i0, type: IonRatingStarsModule, imports: [CommonModule,
        IonicModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.7", ngImport: i0, type: IonRatingStarsModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        IonRatingStarsComponent
                    ],
                    imports: [
                        CommonModule,
                        IonicModule
                    ],
                    exports: [
                        IonRatingStarsComponent
                    ]
                }]
        }] });
