// ionic-rating-component/esm5/lib/ionic-rating.service.js
import { __decorate } from "tslib";
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
var IonicRatingService = /** @class */ (function () {
    function IonicRatingService() {
        this.starRatingSubject = new Subject();
        this.topicSubject = new Subject();
    }
    IonicRatingService.prototype.publishStartRatingChanged = function (value) {
        this.starRatingSubject.next(value);
    };
    IonicRatingService.prototype.getStartRatingChangedObservable = function () {
        return this.starRatingSubject;
    };
    IonicRatingService.prototype.publishTopic = function (topic, value) {
        this.topicSubject.next({ topic: topic, value: value });
    };
    IonicRatingService.prototype.getTopicObservable = function (key) {
        return this.topicSubject[key];
    };
    IonicRatingService.ɵprov = i0.ɵɵdefineInjectable({ factory: function IonicRatingService_Factory() { return new IonicRatingService(); }, token: IonicRatingService, providedIn: "root" });
    IonicRatingService = __decorate([
        Injectable({
            providedIn: 'root'
        })
    ], IonicRatingService);
    return IonicRatingService;
}());
export { IonicRatingService };
